### ISSUE TITLE
HubSpot integration - submission never arrives in Formie but does get saved in Hubspot

### ISSUE THREAD

**Issue by stevefloat** [verbb/formie#1107]
### Describe the bug

We have Hubspot integration set up with our form using the new Private Apps integration. However please note that the issue below also happened via the legacy integration when using API Keys.

Unfortunately the following bug seems to be inconsistent.
We are seeing that a small number of form submissions aren't registering with Formie / Craft. However they are being saved in our Hubspot account.

There is nothing in the logs that accounts for the submission and they don't appear to be spam. They just never show up in Formie Submissions and our Formie notifications aren't triggered.

However the submission and all associated data does arrive in Hubspot.

Most of the time the form all works as intended - data is pushed to Hubspot and the submission is saved in Formie/Craft and our Formie notifications are sent out.

Have you ever seen this before with Hubspot?
It's like it completes the Hubspot part then just silently fails after that.

Are you able to offer any advice or guidance?

### Steps to reproduce

1. Formie form with Hubspot integration (Private Apps)
2. Complete form on website
3. A small number of submissions never make it into Formie/Craft

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

3.7.55.1 PRO

### Plugin version

1.6.13

### Multi-site?

No

**Comment by engram-design** [comment]
So that should be impossible - if they aren't in Formie as submissions, then there's no way Formie should be sending it to HubSpot. My other thought is, are the submissions being marked as spam?

But the nature of the integration is that the submission is saved in Formie first, then sent to HubSpot - not the other way around. As such, it shouldn't be possible that anything is sent to HubSpot if it's not in Formie.

I have seen this issue before with other users, who are using the HubSpot forms embed functionality, or are otherwise using some embed code from HubSpot that means that their form submissions are circumventing Formie altogether. I'll have to ask these users how they resolved it, but I believe disabling any embed code of tracking did the trick.

**Comment by stevefloat** [comment]
Thanks for following up Josh.
The submission aren't spam - I checked that first.

Your reply helps, as it probably means we need to be looking at what Hubspot is doing and perhaps as you say, the submissions are somehow circumventing Formie altogether. 

We do use Hubspot code for live chat. We used to use a Hubspot embedded form before setting up this new site and using Formie. I wonder if Hubspot is trying to be clever - but actually just causing confusion.

If you are able to share details of other users' experiences similar to this, it may help us resolve.

Many thanks.

**Comment by stevefloat** [comment]
Hi Josh - just to follow up in case others find this useful.

It turns out in our Hubspot account settings for non-Hubspot forms we had "Collect data from website forms" turned on:
![image-png-Jun-09-2021-02-03-57-66-AM](https://user-images.githubusercontent.com/52162823/194518664-1fd88ad9-9200-43e0-b3ca-18e79abc340d.png)

Which meant Hubspot was attempting to intercept the form submission and save the data. We then had our own connection to Hubspot via the Formie integration.

So we've turned off Hubspot's "Collect data from website forms" and can confirm Formie is still pushing data in as expected.

It's tricky to test the issue we saw above, as it was so inconsistent. However it does make sense of how the form data possibly got into Hubspot when there was no record in Formie.

For instance, Hubspot say that every time a user presses the submit button it will save a record of the data - even if the form doesn't actually submit due to validation. Which could account for this.

We will monitor and see how this plays out. Thanks again.
